File: Modules/SMES/Tasks/SIE.Web.MES.TaskManagement/SchedulingInfs/Commands/SchedulingInfGenerateTaskCommand.cs
using AngleSharp.Css.Dom;
using SIE.MES.TaskManagement.SchedulingInfs;
using SIE.Web.Command;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SIE.Web.MES.TaskManagement.SchedulingInfs.Commands
{
    /// <summary>
    /// 排程下发(创建派工任务单)
    /// </summary>
    public class SchedulingInfGenerateTaskCommand : ViewCommand<List<double>>
    {
        protected override object Excute(List<double> args, string scope)
        {
            var result = RT.Service.Resolve<SchedulingInfController>().GenerateTask(args);
            if (result.IsNullOrEmpty())
                result = "下发成功!".L10N();
            return result;
        }
    }
}

File: Modules/SMES/Tasks/SIE.Web.MES.TaskManagement/SchedulingInfs/Commands/SchedulingInfGenerateTaskCommand.js
SIE.defineCommand('SIE.Web.MES.TaskManagement.SchedulingInfs.Commands.SchedulingInfGenerateTaskCommand', {
    extend: 'SIE.Web.Common.Import.Commands.ImportCommandBase',
    meta: { text: "手动下发", group: "business", iconCls: "icon-ArrowDown icon-blue" },
    executeIntervalMode: SIE.cmd.IntervalMode.Debounce.value,//使用防抖模式

    canExecute: function (view) {
        //当全部都校验通过了并且不是作废的，才可以下发
        if (view && view.getSelection() && view.getSelection().length > 0 && view.getSelection().all(p => p.getIsCheck() == true && (p.getIsCancel() == false || p.getIsCancel() == null)))
            return true;
        return false;
    },

    execute: function (view) {
        var ids = view.getSelectionIds();
        view.execute({
            data: ids,
            success: function (res) {
                if (res.Success) {
                    view.reloadData();
                    SIE.Msg.showMessage(res.Result);
                }
            }
        });
    }

});